### ISSUE TITLE
Remembering open folders in project tree.

### ISSUE THREAD

**Issue by blitz-research** [engor/Ted2Go#87]
Hi,

It appears that if you close a folder in the project tree but there were subfolders open before you closed it, the folder will be reopened the next time you start ted2go.

For example, if I open modules->std->socket->native, then close 'std', quit ted2 and restart it, modules->std->socket->native are all open again. To *really* close 'std', you need to close 'socket' and 'native' first (I think). IMO this is wrong, and can become quite annoying if you're restarting ted2go a lot, as the project tree appears to be in a random start when you start tedgo.

Basically, if you close a folder it should still be closed when you restart ted2go IMO, even if this means 'forgetting' the state of any subfolders

Perhaps also an option to remember open folders? I am quite happy to have all folders initially closed when I start ted2go.

Probably sounds like I'm nitpicking, but I suspect my usage patterns are pretty unique!

Bye,
Mark

**Comment by engor** [comment]
Hi, it's a bug. I thought I fixed expanding of parent nodes, so I will looking at this again.

**Comment by engor** [comment]
Fixed. Now you will have tree state after restart like it was before closing.
If there are some folders expanded but their parent folders are collapsed - you will see these parents folders collapsed, and you will see expanded child when you expand a parent.
Try it out. :)

**Comment by blitz-research** [comment]
Thanks, works great!


On Mon, Dec 18, 2017 at 8:22 PM, Evgeniy Goroshkin <notifications@github.com
